internal/service/federatedsettingsorgconfig/data_source_federated_settings_connected_orgs.go

```diff
@@ -150,43 +132,36 @@ func dataSourceMongoDBAtlasFederatedSettingsOrganizationConfigsRead(ctx context.
 
 		for i := range federatedSettingsConnectedOrganizations.GetResults() {
-			if federatedSettingsConnectedOrganizations.GetResults()[i].UserConflicts == nil {
-				federatedSettingsConnectedOrganizationsMap[i] = map[string]any{
-					"domain_allow_list":          federatedSettingsConnectedOrganizations.GetResults()[i].GetDomainAllowList(),
-					"domain_restriction_enabled": federatedSettingsConnectedOrganizations.GetResults()[i].GetDomainRestrictionEnabled(),
-					"identity_provider_id":       federatedSettingsConnectedOrganizations.GetResults()[i].GetIdentityProviderId(),
-					"org_id":                     federatedSettingsConnectedOrganizations.GetResults()[i].GetOrgId(),
-					"post_auth_role_grants":      federatedSettingsConnectedOrganizations.GetResults()[i].GetPostAuthRoleGrants(),
-					"role_mappings":              FlattenRoleMappings(federatedSettingsConnectedOrganizations.GetResults()[i].GetRoleMappings()),
-					"user_conflicts":             nil,
-				}
-			} else {
-				federatedSettingsConnectedOrganizationsMap[i] = map[string]any{
-					"domain_allow_list":          federatedSettingsConnectedOrganizations.GetResults()[i].GetDomainAllowList(),
-					"domain_restriction_enabled": federatedSettingsConnectedOrganizations.GetResults()[i].GetDomainRestrictionEnabled(),
-					"identity_provider_id":       federatedSettingsConnectedOrganizations.GetResults()[i].GetIdentityProviderId(),
-					"org_id":                     federatedSettingsConnectedOrganizations.GetResults()[i].GetOrgId(),
-					"post_auth_role_grants":      federatedSettingsConnectedOrganizations.GetResults()[i].GetPostAuthRoleGrants(),
-					"role_mappings":              FlattenRoleMappings(federatedSettingsConnectedOrganizations.GetResults()[i].GetRoleMappings()),
-					"user_conflicts":             FlattenUserConflicts(federatedSettingsConnectedOrganizations.GetResults()[i].GetUserConflicts()),
-				}
+			orgConfig := map[string]any{
+				"domain_allow_list":                 federatedSettingsConnectedOrganizations.GetResults()[i].GetDomainAllowList(),
+				"domain_restriction_enabled":        federatedSettingsConnectedOrganizations.GetResults()[i].GetDomainRestrictionEnabled(),
+				"identity_provider_id":              federatedSettingsConnectedOrganizations.GetResults()[i].GetIdentityProviderId(),
+				"org_id":                            federatedSettingsConnectedOrganizations.GetResults()[i].GetOrgId(),
+				"post_auth_role_grants":             federatedSettingsConnectedOrganizations.GetResults()[i].GetPostAuthRoleGrants(),
+				"role_mappings":                     FlattenRoleMappings(federatedSettingsConnectedOrganizations.GetResults()[i].GetRoleMappings()),
+				"data_access_identity_provider_ids": federatedSettingsConnectedOrganizations.GetResults()[i].GetDataAccessIdentityProviderIds(),
+				"user_conflicts":                    nil,
+			}
+			if federatedSettingsConnectedOrganizations.GetResults()[i].UserConflicts != nil {
```

Comment:
nice cleanup!

Reply:
+1. knit: what about FlattenUserConflicts returning nil if does't exist, so you can just do in the map: 
"user_conflicts":                    FlattenUserConflicts(federatedSettingsConnectedOrganizations.GetResults()[i].GetUserConflicts()),

Reply:
Good point. It was already returning nil :D 
Changed in 62db01d288e9a0d55bdea9f025479df0d88661c5